client/components/tinymce/plugins/wpcom-track-paste/plugin.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import tinymce from 'tinymce/tinymce';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+const debug = debugFactory( 'calypso:tinymce-plugins:wpcom-paste-track' );
+
+const SOURCE_GOOGLE_DOCS = 'google_docs';
+const SOURCE_UNKNOWN = 'unknown';
+
+function trackPaste( editor ) {
+	debug( 'init' );
+
+	const store = editor.getParam( 'redux_store' );
+	if ( ! store ) {
+		debug( 'store not found, do not bind events' );
+		return;
+	}
+
+	/**
+	 * Records the event.
+	 *
+	 * @param {String} mode - 'html-editor' or 'visual-editor', indicates which editor was in use on paste.
+	 * @param {(Array|DOMStringList)} types - The types the content is available to paste.
+	 */
+	function recordPasteEvent( mode, types ) {
+		debug( 'track paste event ' );
+		const typesStr = toString( types );
+		store.dispatch( recordTracksEvent( 'calypso_editor_content_paste', {
+			mode,
+			types: typesStr,
+			source: getSource( typesStr )
+		} ) );
+	}
+
+	function isGoogleDocs( types ) {
+		return ( types.indexOf( 'application/x-vnd.google-docs-image-clip+wrapped' ) > -1 ) ||
+		( types.indexOf( 'application/x-vnd.google-docs-document-slice-clip+wrapped' ) > -1 );
+	}
+
+	function getSource( types ) {
+		if ( isGoogleDocs( types ) ) {
+			return SOURCE_GOOGLE_DOCS;
+		}
+		return SOURCE_UNKNOWN;
```

Comment:
I am expecting that we can discover more sources after this is deployed, so my mind kind of prepared the code for that fact. For now, it is simpler to use ternary operator and let's cross that bridge when/if we get there.